src/Functions/array/arrayAggregation.cpp

```diff
@@ -258,6 +285,10 @@ struct ArrayAggregateImpl
                     res[i] = static_cast<Result>(s) / count;
                 }
             }
+            else if constexpr (aggregate_operation == AggregateOperation::product && IsDecimalNumber<Element>)
+            {
+                res[i] = DecimalUtils::convertTo<Result>(s, data.getScale() * count);
```

Comment:
Why there is `data.getScale() * count` ? Should not there be `data.getScale()` ?